### PR TITLE
fix issue: throw expt when locale is nil

### PR DESCRIPTION
**Pull request by linr211**
fix issue: throw expt when locale is nil

### CODE REVIEW

File: lib/sgtn-client/api/source.rb [hunk 1]
```diff
@@ -15,7 +19,7 @@ def self.getSource(component, key, locale)
         else
           SgtnClient.logger.debug "[Source][getSource]getting sources from cache with key: " + cache_key
         end
-        s = items.nil?? nil : items[locale][key]
+        s = (items.nil? || items[locale].nil?)? nil : items[locale][key]
```
Comment (jessiejuachon) [inline lib/sgtn-client/api/source.rb:22]: This is not correct. Per spec, if source bundle is not found, then it should be loaded right there and then.
Also, in CloudHealth, loadBundles had been invoked at app start up.

**Review by jessiejuachon — CHANGES_REQUESTED** [review]

File: lib/sgtn-client/api/source.rb (hunk 1 above)
Comment (linr211) [reply lib/sgtn-client/api/source.rb:22]: I don't understand the reason, maybe Xiaochao could help explain it. However, from the log, it's this line throw the exception.

Comment (jessiejuachon) [reply lib/sgtn-client/api/source.rb:22]: After adding this, what would the UI display then if no source strings is found?

Comment (linr211) [reply lib/sgtn-client/api/source.rb:22]: It's fixing issues report in the logs, it's saying the array is nil before using it:
undefined method `[]' for nil:NilClass
["/home/cp-web/cloudpercept/shared/bundle/ruby/2.5.0/gems/singleton-client-0.7.4/lib/sgtn-client/api/source.rb:18:in `getSource'"

**Review by Xiaochao8 — DISMISSED** [review]

File: lib/sgtn-client/api/source.rb (hunk 1 above)
Comment (jessiejuachon) [reply lib/sgtn-client/api/source.rb:22]: The array should not be nil in the first place.

**Review by jessiejuachon — DISMISSED** [review]

**Review by Xiaochao8 — APPROVED** [review]

**Review by tigershi — APPROVED** [review]

**Pull request merged**